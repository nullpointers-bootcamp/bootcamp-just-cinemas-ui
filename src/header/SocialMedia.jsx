import React from "react";
import "./socialMedia.css";
import PropTypes from "prop-types";

const SocialMedia = props => (
  <div>
    <a href={props.href}>
      <img className="social-media" src={props.src} />
    </a>
  </div>
);

SocialMedia.defaultProps = {
  href: null,
  src: null
};

SocialMedia.propTypes = {
  href: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired
};

export default SocialMedia;
