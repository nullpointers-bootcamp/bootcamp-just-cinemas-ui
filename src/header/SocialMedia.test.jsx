import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import SocialMedia from "./SocialMedia";

it("show Social media", () => {
  const tree = renderer
    .create(
      <Router>
        <SocialMedia href="https://www.facebook.com/ThoughtWorks/" src="https://s3.ap-south-1.amazonaws.com/twspicinemas/assets/fb_icon.gif" />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
