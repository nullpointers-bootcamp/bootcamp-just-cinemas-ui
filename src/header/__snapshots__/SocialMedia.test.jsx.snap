// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show Social media 1`] = `
<div>
  <a
    href="https://www.facebook.com/ThoughtWorks/"
  >
    <img
      className="social-media"
      src="https://s3.ap-south-1.amazonaws.com/twspicinemas/assets/fb_icon.gif"
    />
  </a>
</div>
`;
